### PR TITLE
Cleanup some nonsymmetric BestEffortReplay usage.

### PR DESCRIPTION
**Pull request by csarofeen**
Try to use `BestEffortReplay::getIterDomainEquivalence` in favor of `BestEffortReplay::getReplay` since the former is a symmetric interface.

### CODE REVIEW

File: torch/csrc/jit/codegen/cuda/compute_at_map.cpp [hunk 1]
```diff
@@ -399,20 +399,28 @@ void IterDomainGraph::build(Fusion* fusion) {
...
-        // consumers even for the Parallel map as they share the same
-        // loop.
-        for (auto c_id : getSortedKeys(c2f_map, Statement::lessThan)) {
-          auto f_id = c2f_map.at(c_id);
-          // Map the id's together
-          permissive_nodes_.mapEntries(f_id, c_id);
-          exact_nodes_.mapEntries(f_id, c_id);
-          if (idIsALeafDomain(f_id, first_output_tv)) {
-            loop_nodes_.mapEntries(f_id, c_id);
+        // consumers
```
Comment (csarofeen) [inline torch/csrc/jit/codegen/cuda/compute_at_map.cpp:403]: Compute at map uses `getReplay` anyways. I'm open minded of leaving this one as `getReplay`.

Comment (naoyam) [reply torch/csrc/jit/codegen/cuda/compute_at_map.cpp:403]: Just want to make sure I understand this correctly, there should be no difference between using `getReplay` and `getIterDomainEquivalence` in this sibling mapping case as there should be no forwarding when replaying from one sibling to another sibling.

**Review by naoyam — APPROVED** [review]
LGTM. Do you intend to deprecate `getReplay`?

**Comment by csarofeen** [comment]
> Just want to make sure I understand this correctly, there should be no difference between using getReplay and getIterDomainEquivalence in this sibling mapping case as there should be no forwarding when replaying from one sibling to another sibling.

Yes.

> LGTM. Do you intend to deprecate getReplay?

That wasn't my goal, and I'm not entirely sure if we can or should. However, I like using the asymmetric interface as little as possible. If we could only use a symmetric interface that would be great, but I'm not sure how much work it is or how much value it would provide.

File: torch/csrc/jit/codegen/cuda/compute_at_map.cpp (hunk 1 above)
Comment (csarofeen) [reply torch/csrc/jit/codegen/cuda/compute_at_map.cpp:403]: Put the response below, but want to put it on here as well so it doesn't look like I didn't respond...

Yes, that's correct.

File: torch/csrc/jit/codegen/cuda/compute_at_map.cpp [hunk 2]
```diff
@@ -399,20 +399,28 @@ void IterDomainGraph::build(Fusion* fusion) {
...
+          }
+        }
+
+        // Map all entries for the Loop map as they share the same loops.
+        for (auto f_id : first_output_tv->domain()->domain()) {
+          auto disjoint_set = c2f_disjoint_sets.getDisjointSetOf(f_id);
+          auto id0 = *(disjoint_set.begin());
+          for (auto id1 : disjoint_set) {
+            loop_nodes_.mapEntries(id0, id1);
           }
```
Comment (zasdfgbnm) [inline torch/csrc/jit/codegen/cuda/compute_at_map.cpp:423]: Will this make `loop_nodes_` contain non-leaf IDs? Is this intentional?

Comment (zasdfgbnm) [reply torch/csrc/jit/codegen/cuda/compute_at_map.cpp:423]: Oh, nvm, this is for mapping siblings, so they should be leaf.

**Review by zasdfgbnm — APPROVED** [review]

**Pull request merged**